lib/smart_proxy_remote_execution_ssh/session.rb

```diff
@@ -30,7 +30,7 @@ def initialize_command
       end
       output_path = File.join(File.dirname(remote_script), 'output')
 
-      @connector.async_run("#{su_prefix}#{remote_script} | /usr/bin/tee #{output_path}") do |data|
+      @connector.async_run("set -o pipefail; #{su_prefix}#{remote_script} | /usr/bin/tee #{output_path}") do |data|
```

Comment:
if its base you can also uses PIPESTATUS, probably can test for the user shell or if it supports pipefail.